pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -883,12 +902,24 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 				if err = controllerutil.MarkPodsNotReady(ctx, nc.kubeClient, nc.recorder, pods, node.Name); err != nil {
 					utilruntime.HandleError(fmt.Errorf("unable to mark all pods NotReady on node %v: %v; queuing for retry", node.Name, err))
 					nc.nodesToRetry.Store(node.Name, struct{}{})
-					continue
+					return
 				}
 			}
 		}
 		nc.nodesToRetry.Delete(node.Name)
 	}
+
+	// Marking the pods not ready on a node requires looping over them and
+	// updating each pod's status one at a time. This is performed serially, and
+	// can take a while if we're processing each node serially as well. So we
+	// process them with bounded concurrency instead, since most of the time is
+	// spent waiting on io.
+	workers := nc.nodeUpdateWorkerSize
+	if workers == 0 {
+		workers = scheduler.UpdateWorkerSize
+	}
```

Comment:
Let's remove it from here and move the initialization of nodeUpdateWorkerSize field to `NewNodeLifecycleController` function.